agent/consul/authmethod/authmethods.go

```diff
@@ -140,7 +141,9 @@ func NewValidator(method *structs.ACLAuthMethod) (Validator, error) {
 		return nil, fmt.Errorf("no auth method registered with type: %s", method.Type)
 	}
 
-	return factory(method)
+	logger = logger.Named("authmethod").With("authMethodType", method.Type, "authMethodName", method.Name)
```

Comment:
```suggestion
	logger = logger.Named("authmethod").With("type", method.Type, "name", method.Name)
```
I kinda like this better

Reply:
i also like this better